lib/libvarnish/vlu.c

```diff
@@ -102,23 +99,22 @@ LineUpProcess(struct vlu *l)
 	if (*p != '\0') {
 		q = strchr(p, '\0');
 		assert(q != NULL);
-		l->bufp = (unsigned)(q - p);
-		memmove(l->buf, p, l->bufp);
-		l->buf[l->bufp] = '\0';
+		VSB_keepsome(l->buf, p, (q - p));
 	} else
-		l->bufp = 0;
+		VSB_clear(l->buf);
 	return (0);
 }
 
 int
 VLU_Fd(int fd, struct vlu *l)
 {
 	int i;
+	char buf[BUFSIZ];
```

Comment:
Actually, I think that `BUFSIZ` is too low here, and that we can afford more than this in each `read` call.